test/Microsoft.IdentityModel.Tokens.Tests/CryptoProviderFactoryTests.cs

```diff
@@ -1130,11 +1130,11 @@ public void ProviderCache_EnsureNoException_MultipleThreads()
         /// Test to ensure no hanging task at the end when calling the JwtSecurityTokenHandler.WriteToken() method.
         /// The JwtHeader is created with SymmetricEncryptingCredentials.
         /// </summary>
-        [Fact (Skip = "test failing")]
+        [Fact]
         public void ProviderCache_EnsureNoLeakingTasks_SecurityTokenHandler_SymmetricEncryptingCredentials()
         {
-            InMemoryCryptoProviderCache cache = CryptoProviderFactory.Default.CryptoProviderCache as InMemoryCryptoProviderCache;
-            SetTaskIdleTimeoutInSeconds(cache, 10); // set the event queue task idle timeout
+            InMemoryCryptoProviderCache cache = new InMemoryCryptoProviderCache();
```

Comment:
This change was necessary in order to get these tests to start passing again, though I'm not sure about the implications of it.

@dannybtsai Do you know why the test was originally written the way it was, and why it might have started failing after the changes in this PR were made?

Reply:
This test was originally written to cover the scenario that the Default CryptoProviderFactory will be referenced (and the event queue task will be started) when clients call the JwtSecurityTokenHandler.WriteToken() method.
(https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/issues/1683)

We need to use the CryptoProviderFactory.Default.CryptoProviderCache instead of creating a new InMemoryCryptoProviderCache as the Default.SymmetricEncryptingCredentials is referencing the CryptoProviderFactory.Default.

Reply:
That makes a lot of sense.... unfortunately these tests are still failing locally for me. 

Just kicked off a topic branch build with this change in place to make sure they pass on the build server.

Reply:
Tests have been re-written to use a new instance of a CryptoProviderFactory; seems other tests were messing with CryptoProviderFactory.Default (adding signature providers, etc.) and that was causing issues.